File: go/grains/grains.go
package grains

import "errors"

const testVersion = 1

func Square(n int) (uint64, error) {
	if n <= 0 || n > 64 {
		return 0, errors.New("Error")
	}
	return 1 << uint(n-1), nil
}

func Total() uint64 {
	var sum uint64 = 0
	for i := 1; i <= 64; i++ {
		n, _ := Square(i)
		sum += n
	}
	return sum
}

File: haskell/grains/src/Grains.hs
module Grains (square, total) where

square :: Integer -> Maybe Integer
square n
  | n >= 1 && n <= 64 = Just (2^(n - 1))
  | otherwise = Nothing

total :: Integer
total = foldr f 0 [0..63]
  where f n sum = sum + 2^n
